backend/stakepoold/rpc/rpcserver/server.go

```diff
@@ -54,32 +59,256 @@ func (*versionServer) Version(ctx context.Context, req *pb.VersionRequest) (*pb.
+	grpcGetIgnoredLowFeeTicketsResultsChan chan map[chainhash.Hash]string
+	grpcGetLiveTicketsResultsChan          chan map[chainhash.Hash]string
+	ticketChan                             chan map[chainhash.Hash]string
+	userChan                               chan map[string]userdata.UserVotingConfig
 }
 
 // StartStakepooldService creates an implementation of the StakepooldService
 // and registers it.
-func StartStakepooldService(c chan struct{}, server *grpc.Server) {
+func StartStakepooldService(
+	getAddedLowFeeTicketsChan chan struct{},
+	getIgnoredLowFeeTicketsChan chan struct{},
+	getLiveTicketsChan chan struct{},
+	grpcGetAddedLowFeeTicketsResultsChan chan map[chainhash.Hash]string,
+	grpcGetIgnoredLowFeeTicketsResultsChan chan map[chainhash.Hash]string,
+	grpcGetLiveTicketsResultsChan chan map[chainhash.Hash]string,
+	ticketChan chan map[chainhash.Hash]string,
+	userChan chan map[string]userdata.UserVotingConfig, server *grpc.Server) {
 	pb.RegisterStakepooldServiceServer(server, &stakepooldServer{
-		c: c,
+		getAddedLowFeeTicketsChan:              getAddedLowFeeTicketsChan,
+		getIgnoredLowFeeTicketsChan:            getIgnoredLowFeeTicketsChan,
+		getLiveTicketsChan:                     getLiveTicketsChan,
+		grpcGetAddedLowFeeTicketsResultsChan:   grpcGetAddedLowFeeTicketsResultsChan,
+		grpcGetIgnoredLowFeeTicketsResultsChan: grpcGetIgnoredLowFeeTicketsResultsChan,
+		grpcGetLiveTicketsResultsChan:          grpcGetLiveTicketsResultsChan,
+		ticketChan:                             ticketChan,
+		userChan:                               userChan,
 	})
 }
 
+func (s *stakepooldServer) GetAddedLowFeeTickets(ctx context.Context, req *pb.GetAddedLowFeeTicketsRequest) (*pb.GetAddedLowFeeTicketsResponse, error) {
+	startTime := time.Now()
+	tickets := make([]*pb.TicketEntry, 0)
+
+	// limit the time we take
+	ctx, cancel := context.WithCancel(ctx)
+	time.AfterFunc(grpcTimeout, cancel)
+
+	log.Debug("GetAddedLowFeeTickets: started")
+
+	// Don't block on messaging.  We want to make sure we can
+	// handle the next call ASAP.
+	waitTime := time.Now()
+	select {
+	case s.getAddedLowFeeTicketsChan <- struct{}{}:
+		log.Debug("GetAddedLowFeeTickets: signaled getAddedLowFeeTicketsChan")
+	default:
+		// We log this in order to detect if we potentially
+		// have a deadlock.
+		log.Debug("GetAddedLowFeeTickets: query not sent")
+		goto done
+	}
+
+	for {
+		select {
+		case addedLowFeeTickets := <-s.grpcGetAddedLowFeeTicketsResultsChan:
+			log.Debug("GetAddedLowFeeTickets: received response from channel")
+			for addedLowFeeTicket, msa := range addedLowFeeTickets {
+				tickets = append(tickets, &pb.TicketEntry{
+					TicketAddress: msa,
+					TicketHash:    addedLowFeeTicket.String(),
+				})
+			}
+			goto done
+		case <-ctx.Done():
+			log.Debug("GetAddedLowFeeTickets: timed out %v", ctx.Err())
+			goto done
+		}
+	}
+
+done:
+	defer func() {
+		log.Infof("GetAddedLowFeeTickets processed %v tickets in %v waitTime %v",
```

Comment:
not sure if this is common in this codebase but i think intercepting the requests (similar to how i do it in wallet) is a much cleaner and simpler way to keep track of this info

Reply:
https://github.com/decred/dcrwallet/blob/768542b7d2dd1d021c51cf3f07c2c24d46151cd3/rpcserver.go#L215

capture the current time before and after running the handler and log the delta

Reply:
using an interceptor would also be a good place to add the deadline to the context